auth-service.js

```diff
@@ -0,0 +1,500 @@
+
+      debug("Clicking ID.me sign-in button from saml-proxy");
+      await Promise.all([
+        page.waitForNavigation({ waitUntil: getWaitUntil() }),
+        page.$eval(".idme-signin", (elem) => elem.click()),
+      ]);
+
+      debug("Looking for expected elements on ID.me sign-in page");
+      await Promise.all([
+        page.waitForSelector("#user_email"),
+        page.waitForSelector("#user_password"),
+        page.waitForSelector('[name="commit"]'),
+      ]);
+
+      debug("Typing username and password on ID.me sign-in page");
+      await page.type("#user_email", args["user-email"]);
+      await page.type("#user_password", args["user-password"]);
+
+      debug("Clicking sign-in button on ID.me sign-in page");
+      await Promise.all([
+        page.waitForNavigation({ waitUntil: getWaitUntil() }),
+        page.$eval('[name="commit"]', (elem) => elem.click()),
+      ]);
+
+      debug("Looking for expected elements on 2FA options page");
+      await Promise.all([
+        page.waitForSelector(".phone"),
+        page.waitForSelector("button.btn-primary"),
+      ]);
+
+      debug("Clicking the send-2FA-code button");
+      await Promise.all([
+        page.waitForNavigation({ waitUntil: getWaitUntil() }),
+        page.$eval("button.btn-primary", (elem) => elem.click()),
+      ]);
+
+      debug("Looking for expected elements on 2FA submit page");
+      await Promise.all([
+        page.waitForSelector("#multifactor_code"),
+        page.waitForSelector("button.btn-primary"),
+      ]);
+
+      debug("Clicking the verify-2FA button");
+      debug("Waiting for next page... expecting consent or final redirect");
+      finalRedirectRequest = await Promise.all([
+        page.waitForRequest(
+          (request) =>
+            request.url().includes(args["redirect-uri"]) ||
+            page.url().includes("signin/consent")
+        ),
+        page.$eval("button.btn-primary", (elem) => elem.click()),
+      ]);
+    }
+
+    if (page.url().includes("signin/consent")) {
+      debug("Waiting for expected elements on consent page to load");
+      await page.waitForSelector("[data-type='save']");
+
+      /*
+       * The consent button will occasionally fail (do nothing) even
+       * though the page is loaded. Injecting a small delay in execution
+       * helps any extra loading necessary to make the button work
+       * more consistently.
+       */
+      debug("Pausing for 1000 ms");
+      await page.waitForTimeout(1000);
+
+      debug("Clicking allow access button on consent page");
+      debug("Waiting for final redirect");
+      finalRedirectRequest = await Promise.all([
+        page.waitForRequest((request) =>
+          request.url().includes(args["redirect-uri"])
+        ),
+        page.$eval("[data-type='save']", (elem) => elem.click()),
+      ]);
+    }
+
+    finalRedirectRequest = finalRedirectRequest[0];
+  } catch (error) {
+    return getError(lastDebugMessage, error, page);
+  }
+
```

Comment:
This script seems to not exit because Puppeteer keeps running because `browser.close()` is never called. This isn't a problem for the CLI because `process.exit()` is called. But that seems to be why the lib function keeps running if `process.exit()` is not called

```suggestion

  browser.close()

```

Reply:
This `browser.close()` is still missing. It also needs to be called when the function exits successfully